lib/sync/change.ts

```diff
@@ -100,19 +96,32 @@ export interface SyncChanges {
 export function calculateChanges(
     before: k8s.KubernetesObject[],
     after: k8s.KubernetesObject[] | undefined,
-    change: "apply" | "delete",
+    change: ChangeType,
 ): SyncChanges[] {
 
-    if (change === "delete") {
-        return (before || []).map(spec => ({ change, spec }));
-    }
-    const changes: SyncChanges[] = (after || []).map(spec => ({ change, spec }));
+    const changes: SyncChanges[] = (after || []).filter(spec => !hasMetadataAnnotation(spec, "ignore")).map(spec => ({ change, spec }));
     if (before && before.length > 0) {
         for (const spec of before) {
-            if (!after.some(a => sameObject(a, spec))) {
+            if (hasMetadataAnnotation(spec, "ignore")) {
+                changes.push({ change: "ignore", spec });
+            } else if ((change === "delete") || (!after.some(a => sameObject(a, spec)))) {
```

Comment:
What do you expect the behavior to be if the annotation is added or removed as part of the change?

Reply:
If the annotation is added then it takes effect immediately - the change will not be applied/deleted.

If the annotation is removed then it will take a second commit to trigger an apply/delete.

Reply:
This would all be a lot more clear to me if you filtered out specs with "ignore" at the start and then the logic just remained the same. Perhaps if there were a tests that demonstrated the behavior of a spec adding and removing the ignore annotation.